app/assets/javascripts/discourse/app/components/user-menu/bookmark-item.js

```diff
@@ -0,0 +1,47 @@
+import UserMenuItem from "discourse/components/user-menu/menu-item";
+import { NO_REMINDER_ICON } from "discourse/models/bookmark";
+import { postUrl } from "discourse/lib/utilities";
+
+export default class UserMenuBookmarkItem extends UserMenuItem {
+  get className() {
+    return "bookmark";
+  }
+
+  get linkHref() {
+    if (["Topic", "Post"].includes(this.bookmark.bookmarkable_type)) {
```

Comment:
Not really related to your PM but I prefer if the server side was shipping down an attribute to indicate this instead of the client having to figure this out.

Reply:
Ok, done. I went with a new serializers attribute called `url_for_ui`.